### DISCUSSION TITLE
Can anyone teach me how to debug the code in oxc_linter?

### DISCUSSION THREAD

**Discussion by woai3c** [oxc-project/oxc#2923] [Q&A] [answered]
Can anyone teach me how to debug the code in oxc_linter? For example, if I want to debug the code in `crates\oxc_linter\src\rules\eslint\max_lines.rs`, how can I execute this file? I tried to build the code and run some examples, but it seems that I can't directly execute the file in `crates\oxc_linter\src\rules\eslint\max_lines.rs`. My current vscode/launch.json is as follows:
```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "lldb",
            "request": "launch",
            "name": "Debug oxc_linter",
            "program": "${workspaceFolder}/target/debug/examples/linter",
            "args": [],
            "cwd": "${workspaceFolder}"
        }
    ]
}
```
I wonder if anyone can help solve this problem, thank you.

**Comment by mysteryven** [comment]
Make sure to install the [rust-analyzer](https://marketplace.visualstudio.com/items?itemName=rust-lang.rust-analyzer) in your VSCode, the `Debug` button should show above your test function:

<img width="1445" alt="image" src="https://github.com/oxc-project/oxc/assets/33973865/4fee45f3-5bdb-43fe-9502-c86515a72edb">

**Reply by woai3c** [reply, marked as answer]
Thank you your reply.
